test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/Utility/AuditUtilityTests.cs

```diff
@@ -179,6 +179,9 @@ public async Task Check_ProjectReferencingPackageWithVulnerability_WarningLogged
 
         context.PackagesDependencyProvider.Package("pkga", "1.0.0").DependsOn("pkgb", "1.0.0");
         context.PackagesDependencyProvider.Package("pkgb", "1.0.0");
+        context.PackageDownloadPackages.Add(new DownloadDependency("pkgDownload", new VersionRange(new NuGetVersion("1.0.0"), maxVersion: new NuGetVersion("1.0.0"))));
```

Comment:
more of a nitpick:  these package download package lines look a lot less nice than adding the package reference packages on the two lines above, since they're using a builder pattern, but this package download is not.

I wonder if it's worthwhile creating a `DownloadDependencyProvider` so tests that use it are easier to read.

Having said that, half the line length could be reduced if the `VersionRange` was changed to `VersionRange.Parse("[1.0.0])`.  I wouldn't be surprised if I did the `new VersionRange` thing in the past, since I know it will execute faster than Parse. I keep changing my mind between "OMG, NuGet's tests are so slow, we need to make them faster, if only by a little bit" and also "these tests need to be easier to read" 🤣

Reply:
I don't think we ever sort the log messages. 
The order in which they're raised is preserved, which is why we've never had issues. 
They're never relevant in the input/output calculation, since they're normally just be replayed and there's per log code sorting in things like error list and msbuild.

Reply:
Thanks for pointing this out. Personally, I am indifferent on this. But `VersionRange.Parse()` looks cleaner to me